src/components/relationships.tsx

```diff
@@ -47,7 +48,60 @@ export const RelationshipsTab = (
 ) => {
   const { datasource, query, onQueryChange, queryBinder } = props;
   const route = queryBinder ? `${queryBinder}.${queryTypeSelector}` : `${queryTypeSelector}`;
-
+  const [options, setOptions] = useState([]);
+  const updateOptions = async () => {
+    const options = await datasource.relationshipsDatasource.getSourceExternalIds(
+      _.get(query, route)
+    );
+    setOptions(options);
+  };
+  const resetDepth = () => onQueryChange(_.set(query, `${route}.depth`, 1));
+  const handleSelectedItem = useCallback(
+    ({ nodes }) => {
+      const sourceExternalIds = _.uniq(
+        _.map(
+          _.filter(options, ({ sourceExternalId }) =>
+            _.includes(
+              _.map(nodes, (item) => _.last(item.split('-'))),
+              sourceExternalId
+            )
+          ),
+          'sourceExternalId'
+        )
+      );
+      onQueryChange(_.set(query, `${route}.sourceExternalIds`, sourceExternalIds));
+    },
+    [options]
+  );
+  const eventsSubscribe = async () => {
+    const appEvents = await appEventsLoader;
+    appEvents.on(visNodeGraphPanelClickEvent, handleSelectedItem);
+  };
+  const eventsUnsubscribe = async () => {
+    const appEvents = await appEventsLoader;
+    appEvents.off(visNodeGraphPanelClickEvent, resetSource);
```

Comment:
datasource is listening to a panel plugin events? is there any better way to solve this, so we don't have to couple write a plugin-specific logic?

Reply:
At first it felt to be a good idea to use the grafan-event-handler to query click event from the  https://github.com/cognitedata/cognite-grafana-relationships-visualization plugin when a node was selected. We can take this out and maybe is there a "better" way, to handled the click event, as I remember it were problems to call the datasource(s) in a plugin project. I assume, in this phase is better to take this event-listener out and try other scenarios? Please make a decision how to perform forward.